### PR TITLE
recombine: add timeout configuration option

### PR DESCRIPTION
**Pull request by pmalek-sumo**

### CODE REVIEW

**Comment by pmalek-sumo** [comment]
Performance-wise this doesn't show any signs of a regression:

```
go test -v -count 1 -run ^$ -bench Benchmark -benchmem -count 4 ./... # with patch
goos: darwin
goarch: amd64
pkg: github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/recombine
cpu: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
BenchmarkRecombine
BenchmarkRecombine-16             419072              2894 ns/op             280 B/op         10 allocs/op
BenchmarkRecombine-16             395395              2849 ns/op             280 B/op         10 allocs/op
BenchmarkRecombine-16             410451              2874 ns/op             280 B/op         10 allocs/op
BenchmarkRecombine-16             436786              2874 ns/op             280 B/op         10 allocs/op
PASS
ok      github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/recombine   6.057s

go test -v -count 1 -run ^$ -bench Benchmark -benchmem -count 4 ./... # without patch
goos: darwin
goarch: amd64
pkg: github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/recombine
cpu: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
BenchmarkRecombine
BenchmarkRecombine-16             423274              2872 ns/op             280 B/op         10 allocs/op
BenchmarkRecombine-16             434774              2834 ns/op             280 B/op         10 allocs/op
BenchmarkRecombine-16             432498              2845 ns/op             280 B/op         10 allocs/op
BenchmarkRecombine-16             437214              2847 ns/op             280 B/op         10 allocs/op
PASS
ok      github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/recombine   6.223s
```

File: operator/builtin/transformer/recombine/recombine.go [hunk 1]
```diff
@@ -123,12 +128,31 @@ type RecombineOperator struct {
...
+					r.Errorw("Failed flushing", "error", err)
+				}
+				r.Unlock()
+			case <-r.chClose:
+				r.ticker.Stop()
+				return
+			}
+		}
+	}()
+
```
Comment (djaglowski) [inline operator/builtin/transformer/recombine/recombine.go:155]: Can I ask you to flatten this somewhat, perhaps by pulling out a function?

**Review by djaglowski — COMMENTED** [review]
@pmalek-sumo This looks good to me except I wonder if we should name this parameter `force_flush_period`, since it is basically the same as is done in [file_input](https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/operators/file_input.md#configuration-fields). What do you think?

File: operator/builtin/transformer/recombine/recombine.go (hunk 1 above)
Comment (pmalek-sumo) [reply operator/builtin/transformer/recombine/recombine.go]: @djaglowski I've pulled that out to a `func`. Let me know if it's better now.

**Review by rockb1017 — APPROVED** [review]

**Comment by djaglowski** [comment]
> @pmalek-sumo This looks good to me except I wonder if we should name this parameter `force_flush_period`, since it is basically the same as is done in [file_input](https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/operators/file_input.md#configuration-fields). What do you think?

@pmalek-sumo What do you think about this suggestion?

**Comment by pmalek-sumo** [comment]
> > @pmalek-sumo This looks good to me except I wonder if we should name this parameter `force_flush_period`, since it is basically the same as is done in [file_input](https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/operators/file_input.md#configuration-fields). What do you think?
> 
> @pmalek-sumo What do you think about this suggestion?

I've missed that somehow. Seems reasonable. Let me rename this

**Comment by pmalek-sumo** [comment]
> > @pmalek-sumo This looks good to me except I wonder if we should name this parameter `force_flush_period`, since it is basically the same as is done in [file_input](https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/operators/file_input.md#configuration-fields). What do you think?
> 
> @pmalek-sumo What do you think about this suggestion?

Done, PTAL @djaglowski

**Review by djaglowski — APPROVED** [review]

**Pull request merged**